### PR TITLE
Misc Packet Cleanup

### PR DESCRIPTION
**Pull request by sappenin**
* Revert InterledgerAddress.value to `getValue`.
* Add `InterledgerAddressPrefix` that can be an address or allocation scheme, plus unit test.
* Add Codec for `InterledgerAddressPrefix`, plus unit tests.
* Move `data` to `InterlederPacket` parent interface.
* Set default value of Prepare packet to ZERO.
* Add Javadoc to InterledgerFulfillPacket.
* Make BtpSubProtocolContentTypeTest use the proper type (Short).

Signed-off-by: sappenin <sappenin@gmail.com>

### CODE REVIEW

File: ilp-core/src/main/java/org/interledger/core/InterledgerAddressPrefix.java [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+   *
+   * <p>This immutable is interned because it only holds a {@link String} value, which itself is
+   * interned via the Java String pool.</p>
+   */
+  @Value.Immutable
+  @Value.Style(
+      typeBuilder = "*Builder",
+      visibility = Value.Style.ImplementationVisibility.PRIVATE,
+      builderVisibility = Value.Style.BuilderVisibility.PUBLIC,
+      defaults = @Value.Immutable(intern = true))
```
Comment (sublimator) [inline ilp-core/src/main/java/org/interledger/core/InterledgerAddressPrefix.java]: Oooooh! Didn't know about the intern feature

File: ilp-core/src/main/java/org/interledger/core/InterledgerProtocolException.java [hunk 2]
```diff
@@ -35,20 +35,20 @@
...
-   * Constructs a new Interledger protocol exception with the specified reject packet and
-   * detail message.
+   * Constructs a new Interledger protocol exception with the specified reject packet and detail
+   * message.
    *
    * @param interledgerRejectPacket An instance of {@link InterledgerRejectPacket} that is the
-   *                                 underlying error encapsulated by this exception.
-   * @param message The detail message.
+   *                                underlying error encapsulated by this exception.
+   * @param message                 The detail message.
```
Comment (sublimator) [inline ilp-core/src/main/java/org/interledger/core/InterledgerProtocolException.java]: infi-nit:
```
@param message                 The detail message.
```

File: ilp-core/src/main/java/org/interledger/core/InterledgerAddressPrefix.java [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+        return Error.ILLEGAL_ENDING.getMessageFormat();
+      }
+
+      final List<String> schemeAndSegments = Arrays.asList(
+          invalidAddressString.split(SEPARATOR_REGEX, -1)
+      );
+      // validates scheme prefix format
+      final String schemePrefix = schemeAndSegments.get(0);
+      if (!Pattern.compile(SCHEME_REGEX).matcher(schemePrefix).matches()) {
+        return String.format(Error.INVALID_SCHEME_PREFIX.getMessageFormat(), schemePrefix);
```
Comment (sublimator) [inline ilp-core/src/main/java/org/interledger/core/InterledgerAddressPrefix.java]: No win in precompiling the regexes ?

**Comment by sublimator** [comment]
I had a quick (tired) look over and didn't see anything glaring

File: ilp-core/src/main/java/org/interledger/core/InterledgerProtocolException.java (hunk 2 above)
Comment (sappenin) [reply ilp-core/src/main/java/org/interledger/core/InterledgerProtocolException.java]: I think the spacing is purposeful -- it aligns with the description of the `message` field.

**Comment by sublimator** [comment]
Doesn’t matter either way :)

File: ilp-core/src/main/java/org/interledger/core/InterledgerAddressPrefix.java (hunk 3 above)
Comment (sappenin) [reply ilp-core/src/main/java/org/interledger/core/InterledgerAddressPrefix.java]: That's a great point - this code was part of an external submission in `InterledgerAddress` that I mostly just adapted here. I'll try that out.

Comment (sappenin) [reply ilp-core/src/main/java/org/interledger/core/InterledgerAddressPrefix.java]: Fixed.

**Pull request merged**